Engine/Alphas/StatisticsInsightManagerExtension.cs

```diff
@@ -99,16 +104,20 @@ public void OnInsightClosed(InsightAnalysisContext context)
             var volume = _tradablePercentOfVolume * context.InitialValues.Volume;
 
             // value of the entering the trade in the account currency
-            var enterValue = volume * context.InitialValues.Price * context.InitialValues.QuoteCurrencyConversionRate;
+            var enterValue = context.InitialValues.Price * context.InitialValues.QuoteCurrencyConversionRate;
 
             // value of exiting the trade in the account currency
-            var exitValue = volume * context.CurrentValues.Price * context.CurrentValues.QuoteCurrencyConversionRate;
+            var exitValue = context.CurrentValues.Price * context.CurrentValues.QuoteCurrencyConversionRate;
 
             // total value delta between enter and exit values
             var insightValue = (int)context.Insight.Direction * (exitValue - enterValue);
```

Comment:
We want to measure the magnitude change rather than total value delta for kelly criterion, so the value that needs to be added in `_kellyCriterionManager.AddNewValue(insightValue, context.Insight.GeneratedTimeUtc);` should be
```
(int)context.Insight.Direction * ((exitValue / enterValue) - 1);
```

This is the calculation in python, where all three are columns from a pandas DF, Reference Value Final is the stock price at Insight close, and Reference Value is the stock price at Insight creation.
```scores = df['Direction'].values * ((df['Reference Value Final'].values / df['Reference Value'].values) - 1)```

Reply:
Good catch, PR updated 👍